### PR TITLE
docs: add "Customizing Kubo" doc

### PR DESCRIPTION
**Pull request by guseggert**

### CODE REVIEW

**Comment by BigLep** [comment]
Thanks for starting this Gus!  I think a summary table of pros/cons of the options could be useful too (after have the text in place)

**Comment by BigLep** [comment]
2023-05-05 maintainer conversation: lets also update https://github.com/ipfs/kubo/tree/master/docs/examples/kubo-as-a-library as part of this

**Comment by guseggert** [comment]
> 2023-05-05 maintainer conversation: lets also update https://github.com/ipfs/kubo/tree/master/docs/examples/kubo-as-a-library as part of this

Do you remember what specifically we wanted to update in there?

**Comment by BigLep** [comment]
> > 2023-05-05 maintainer conversation: lets also update https://github.com/ipfs/kubo/tree/master/docs/examples/kubo-as-a-library as part of this
> 
> @BigLep Do you remember what specifically we wanted to update in there?

@guseggert : I don't know exactly.  @aschmahmann may have other thoughts.  I think at the minimum it was to signpost from that document to the new one you have created helping make clear that there are other (and maybe better) ways to proceed than using Kubo as a library.

File: docs/customizing.md
Comment (BigLep) [inline docs/customizing.md:16]: Idea: turn the headers into anchor links to the sections below.

File: docs/customizing.md
Comment (BigLep) [inline docs/customizing.md:53]:
```suggestion
Kubo currently works with Go plugins. But using Go plugins requires that you compile the plugin using the exact same version of the Go toolchain with the same configuration (build flags, environment variables, etc.). As a result, you likely need to build Kubo and the plugins together at the same time, and at that point you may as well just compile the functionality directly into Kubo and avoid Go plugins.
```

**Review by BigLep — COMMENTED** [review]
Thanks for writing @guseggert .

How much does the document change when https://github.com/ipfs/kubo/pull/9782 is added?

I leave it to others who know more of the details to review/approve.

**Comment by guseggert** [comment]
> How much does the document change when https://github.com/ipfs/kubo/pull/9782 is added?

We'd add a column to the table and a section describing it

**Review by hacdias — APPROVED** [review]

**Comment by hacdias** [comment]
I am merging this once CI is ✅. Once #9782 is completed, this table should be updated (preferably in the PR itself).

**Pull request merged**
